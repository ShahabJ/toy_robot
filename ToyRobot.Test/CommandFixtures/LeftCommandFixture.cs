using System;
using System.Collections.Generic;
using ApprovalTests;
using ApprovalTests.Reporters;
using FluentAssertions;
using ToyRobot.Library.Commands;
using ToyRobot.Library.Model;
using ToyRobot.Library.Service;
using Xunit;

namespace ToyRobot.Test
{
    [UseReporter(typeof(DiffReporter))]
    public class LeftCommandFixture
    {
        private CommandBase command;
        private GenericRobot robot;
        private GenericTable table;
        public LeftCommandFixture()
        {
            command = new LeftCommand();
            table = new GameTable();
            robot = new GameRobot(table);
        }

        [Fact]
        public void IsValidCommandReport()
        {
            var commands =new String[]{"Left", "left", "LEFT", "LEFT ", "LEFT 1", " LEFT" ,"LFEL"};
            var result = new List<string>();

            foreach (var item in commands)
            {
                var msg = $"{item} ==> {command.IsValidCommand(item)}";
                result.Add(msg);
            }
            Approvals.VerifyAll(result,string.Empty);
        }

        [Fact]
        public void ExecuteReport()
        {   
            var result = new List<string>();
            var positions = new List<Position>();
            
            positions.Add(new Position(0, 0, DirectionEnum.NORTH));
            positions.Add(new Position(0, 0, DirectionEnum.EAST));
            positions.Add(new Position(0, 0, DirectionEnum.WEST));
            positions.Add(new Position(0, 0, DirectionEnum.SOUTH));

            foreach (var position in positions)
            {
                robot.CurrentPosition = position;
                string currentPosition = robot.CurrentPosition.ToString();
                string newPosition = command.Execute(robot).CurrentPosition.ToString();
                result.Add($"Position before TurnLeft:{currentPosition}, Position After TurnLeft:{newPosition}");
            }

            Approvals.VerifyAll(result,"Verifyinng Left Commnad => ");
        }
    }
}
