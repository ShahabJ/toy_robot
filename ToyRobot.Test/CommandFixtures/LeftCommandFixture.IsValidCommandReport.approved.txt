[0] = Left ==> True
[1] = left ==> True
[2] = LEFT ==> True
[3] = LEFT  ==> False
[4] = LEFT 1 ==> False
[5] =  LEFT ==> False
[6] = LFEL ==> False
